### PR TITLE
Fix cloudhut/common having no JSON tags

### PR DESCRIPTION
**Pull request by pvsune**
[cloudhut/common](https://github.com/cloudhut/common) types has no JSON tags.  This PR removes dependency from cloudhut in the API definition by grabbing those type definitions:
- `rest.Config` - Added JSON tags. We can have this changes merged upstream and we have no need to grab these.
- `logging.Config` - Omitted `LogLevel` because `zap.AtomicLevel` is not JSON serializeable.

This PR also introduces aliases for type definitions:
- `console.Config`
- `connect.Config`
- `msgpack.Config`
- `proto.Config`

as controller-gen `make generate` does not create `DeepCopyInto()`.
REF https://github.com/kubernetes-sigs/controller-tools/issues/583

Note:
- We need to `make generate` and `make manifests` for other PRs after this merge.

### CODE REVIEW

File: src/go/k8s/apis/redpanda/v1alpha1/console_types.go [hunk 1]
```diff
@@ -42,6 +38,25 @@ type ConsoleSpec struct {
 	Version string `json:"version"`
 }
 
+// Config for a HTTP server
+// Grabbed from https://github.com/cloudhut/common
+// Added JSON tags. We can have this merged from upstream.
+type RESTConfig struct {
+	ServerGracefulShutdownTimeout time.Duration `yaml:"gracefulShutdownTimeout" json:"gracefulShutdownTimeout"`
+
+	HTTPListenAddress      string        `yaml:"listenAddress" json:"listenAddress"`
```
Comment (nicolaferraro) [inline src/go/k8s/apis/redpanda/v1alpha1/console_types.go:47]: I've not followed the whole conversation. Are you going with representing the full console config as CRD?

I think the approach is good, especially until the api is alpha or console does not make backward incompatible changes.

File: src/go/k8s/apis/redpanda/v1alpha1/console_types.go [hunk 2]
```diff
@@ -10,26 +10,22 @@
...
 
 // ConsoleSpec defines the desired state of Console
 type ConsoleSpec struct {
-	Console console.Config  `json:"console"`
-	Connect connect.Config  `json:"connect"`
-	REST    rest.Config     `json:"server"`
-	Logger  *logging.Config `json:"logger,omitempty"`
+	Console  ConsoleConfig `json:"console"`
+	Connect  ConnectConfig `json:"connect"`
+	REST     RESTConfig    `json:"server"`
```
Comment (nicolaferraro) [inline src/go/k8s/apis/redpanda/v1alpha1/console_types.go:24]: Nit: it would have been more readable if split in a different commit

File: src/go/k8s/apis/redpanda/v1alpha1/console_common.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+// ProtoConfig is a redefinition of console.ProtoConfig but struct fields defined locally
+type ProtoConfig struct {
+	Enabled bool `json:"enabled"`
+
+	// The required proto definitions can be provided via SchemaRegistry, Git or Filesystem
+	SchemaRegistry proto.SchemaRegistryConfig `json:"schemaRegistry"`
+	Git            GitConfig                  `json:"git"`
+	FileSystem     FilesystemConfig           `json:"fileSystem"`
```
Comment (nicolaferraro) [inline src/go/k8s/apis/redpanda/v1alpha1/console_common.go:46]: How does it work in a k8s environment? Are users allowed to mount volumes?

File: src/go/k8s/apis/redpanda/v1alpha1/console_types.go (hunk 1 above)
Comment (pvsune) [reply src/go/k8s/apis/redpanda/v1alpha1/console_types.go:47]: I think Rafal went on implementing the whole Console config as CRD for flexibility in configuration. We may or may not need all of them though.

File: src/go/k8s/apis/redpanda/v1alpha1/console_common.go (hunk 3 above)
Comment (pvsune) [reply src/go/k8s/apis/redpanda/v1alpha1/console_common.go:46]: Right now you cannot specify this in Console CRD but technically users can update volumeMounts of the deployment. I think we are going with representing as much of the Console config as we can for flexibility. https://github.com/redpanda-data/console/blob/master/docs/config/console.yaml#L85
